src/test/java/dev/morling/kccli/util/ConfigurationContextTest.java

```diff
@@ -0,0 +1,114 @@
+/*
+ *  Copyright 2021 The original authors
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package dev.morling.kccli.util;
+
+import dev.morling.kccli.service.Context;
+import org.junit.jupiter.api.DisplayNameGeneration;
+import org.junit.jupiter.api.DisplayNameGenerator.ReplaceUnderscores;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.nio.file.Files;
+import java.util.stream.Stream;
+
+import static net.javacrumbs.jsonunit.assertj.JsonAssertions.assertThatJson;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.params.provider.Arguments.arguments;
+
+@DisplayNameGeneration(ReplaceUnderscores.class)
+class ConfigurationContextTest {
+    @TempDir
+    File tempDir;
+
+    @Nested
+    class SetConfiguration {
+        @MethodSource("dev.morling.kccli.util.ConfigurationContextTest#setConfigurationArguments")
+        @ParameterizedTest
+        void should_create_a_new_configuration_when_no_configuration_exists(String contextName, Context context, String expectedConfiguration) throws IOException {
+            System.setProperty("user.home", tempDir.getAbsolutePath());
```

Comment:
Neat trick :) Alternatively, there could be a constructor in `ConfigurationContext` which takes the config directory, while the default constructor would use the home dir.

Reply:
Had the exact same idea, but didn't want to change too much code at first. Good suggestion!

Reply:
Done.